packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -406,23 +387,6 @@ class EmberRouter extends EmberObject.extend(Evented) implements Evented {
         newInfos: PrivateRouteInfo[],
         transition: Transition
       ) {
-        if (ROUTER_EVENTS) {
-          if (router.willTransition !== defaultWillTransition) {
-            deprecate(
-              'You attempted to override the "willTransition" method which is deprecated. Please inject the router service and listen to the "routeWillChange" event.',
-              false,
-              {
-                id: 'deprecate-router-events',
-                until: '4.0.0',
-                url: 'https://deprecations.emberjs.com/v3.x#toc_deprecate-router-events',
-                for: 'ember-source',
-                since: {
-                  enabled: '3.11.0',
-                },
-              }
-            );
-          }
-        }
         router.willTransition(oldInfos, newInfos, transition);
```

Comment:
Is this the correct way to deal with this? Or do these methods go away entirely?